src/TagHelperPack/AuthzTagHelper.cs

```diff
@@ -85,23 +101,45 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
         }
         else if (!string.IsNullOrEmpty(RequiredPolicy))
         {
-            // auth-policy="foo" & user is authorized for policy "foo"
-            var cacheKey = AspAuthzPolicyAttributeName + "." + RequiredPolicy;
             bool authorized;
-            var cachedResult = ViewContext.ViewData[cacheKey];
-            if (cachedResult != null)
+            if (!string.IsNullOrEmpty(RequiredPolicyPermission))
             {
-                authorized = (bool)cachedResult;
+                // auth-policy="foo" & user is authorized for policy "foo" based on permission
+                var cacheKey = AspAuthzPolicyPermissionAttributeName + "." + RequiredPolicyPermission;
+                var cachedResult = ViewContext.ViewData[cacheKey];
+                if (cachedResult != null)
+                {
+                    authorized = (bool)cachedResult;
+                }
+                else
+                {
+                    var authResult = await _authz.AuthorizeAsync(user, RequiredPolicyPermission, RequiredPolicy);
+                    authorized = authResult.Succeeded;
+                    ViewContext.ViewData[cacheKey] = authorized;
+                }
             }
             else
             {
-                var authResult = await _authz.AuthorizeAsync(user, ViewContext, RequiredPolicy);
-                authorized = authResult.Succeeded;
-                ViewContext.ViewData[cacheKey] = authorized;
+                // auth-policy="foo" & user is authorized for policy "foo"
+                var cacheKey = AspAuthzPolicyAttributeName + "." + RequiredPolicy;
+                var cachedResult = ViewContext.ViewData[cacheKey];
+                if (cachedResult != null)
+                {
+                    authorized = (bool)cachedResult;
+                }
+                else
+                {
+                    var authResult = await _authz.AuthorizeAsync(user, ViewContext, RequiredPolicy);
+                    authorized = authResult.Succeeded;
+                    ViewContext.ViewData[cacheKey] = authorized;
+                }
             }
-
             showOutput = authorized;
         }
+        else if (!string.IsNullOrEmpty(RequiredRole))
```

Comment:
Add a guard to ensure invalid combinations of attributes aren't being set and throw an `InvalidOperationException` in that case, e.g. if both a required role and policy are set.